Adds --db to neo4j-admin memrec

tinwelint opened the pull request:
Which will print a minimal page cache memory setting for the
selected database in addition to the generic memory distribution
based on the machine specs

tinwelint committed:
Adds --db to neo4j-admin memrec

Which will print a minimal page cache memory setting for the
selected database in addition to the generic memory distribution
based on the machine specs

chrisvest commented on community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java:
```diff
@@ -118,7 +131,12 @@ public static Arguments buildArgs()
+                        "Specific database to calculate page cache memory requirement for. " +
+                        "The generic calculation is still a good generic recommendation for this machine, " +
+                        "but there will be an additional calculation for minimal required page cache memory " +
+                        "for mapping all store and index files that are managed by the page cache." ) );
```
I think the standard way of instructing a command to work on a database, is to use the `withDatabase` argument. I think we should do that here as well, instead of assuming that the configured default database is the one we want to build a configuration for in the particular invocation of the command.

Unfortunately, `withDatabase` does not allow you to override the description. I think that should be added, because the description you have here is really good.

chrisvest replied:
The `ConsistencyCheckCommand.execute` method has an example of using `arguments.has( "database" )` to check whether the `--database` argument was specified on the command line or not.

chrisvest commented on community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java:
```diff
@@ -199,6 +219,92 @@ public void execute( String[] args ) throws IncorrectUsage
+
+        print( "#" );
+        print( "# A minimal page cache memory setting to fully map the database contents of '" + storeDir.getAbsolutePath() + "':" );
+        print( pagecache_memory.name() + "=" + dbSpecificPageCacheMemory( storeDir ) );
```
Since this output is kind of meant to the copy-pasted more-or-less directly into `neo4j.conf`, I think we should only output one active `dbms.memory.pagecache.size`. So what I think should happen when `--database` is enabled, is that the `print( pagecache_memory.name() + "=" + pagecache );` line above should actually print the setting as commented out, like `print( "# " + pagecache_memory.name() + "=" + pagecache );`.

chrisvest commented on community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java:
```diff
@@ -199,6 +219,92 @@ public void execute( String[] args ) throws IncorrectUsage
+                    total += outsideWorld.fileSystem().getFileSize( file );
+                }
+            }
+        }
```
Small nit, but this could also be a separate `sumStoreFiles` method, as a sibling to `sumNativeIndexFiles`.

chrisvest commented on community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java:
```diff
@@ -199,6 +219,92 @@ public void execute( String[] args ) throws IncorrectUsage
+
+        // All native index files
+        File baseSchemaIndexDir = baseSchemaIndexFolder( storeDir );
+        total = sumNativeIndexFiles( baseSchemaIndexDir, getNativeIndexFileFilter(), total );
```
This could be `total += …` and then there's no need to pass `total` as an argument to the method.

chrisvest reviewed (commented):
I had a couple of comments.

tinwelint commented:
Thanks. Also marking as not ready for merge because the directory structure for temporal/spatial in lucene-1.0 has a planned change ahead.

tinwelint committed:
Allows neo4j-admin 'database' argument w/ custom description

tinwelint committed:
Better output of memrec command for specific db

- Uses the standard --database <dbName> with specific description.
- Outputs the generic hint as commented out
- Better testing, including all various index providers

tinwelint commented:
OK @chrisvest have a look now

chrisvest commented on community/dbms/src/main/java/org/neo4j/commandline/dbms/Yoyo.java:
```diff
@@ -0,0 +1,59 @@
+import static org.neo4j.values.storable.LocalTimeValue.localTime;
+import static org.neo4j.values.storable.Values.pointValue;
+
+public class Yoyo
```
Looks like you added this by accident.

tinwelint replied:
What `Yoyo` file? :D

chrisvest reviewed (changes requested):
Just one thing…
